lib/core/utils/context.ext.dart

```diff
@@ -0,0 +1,11 @@
+part of '../core.dart';
+
+extension ContextX on BuildContext {
+  ThemeData themeFromPrimaryColor(Color primaryColor) {
+    return Theme.of(this).copyWith(
+      colorScheme: Theme.of(this).colorScheme.copyWith(
+            primary: primaryColor,
+          ),
+    );
+  }
+}
```

Comment:
I feel like this is quite a hack no? We can just give the theme to MaterialApp and this should not be needed, right?

Reply:
Let me test with directly setting it and see if it affects the app.